lib/phoenix_live_view/test/client_proxy.ex

```diff
@@ -356,6 +356,18 @@ defmodule Phoenix.LiveViewTest.ClientProxy do
     {:noreply, state}
   end
 
+  def handle_call({:open_browser, open_browser_fun, html}, from, state) do
+    filename = "#{state.root_view.topic}_#{Phoenix.LiveView.Utils.random_id()}"
+    html = DOM.maybe_wrap_html_element(html, state.root_view.endpoint.static_path("/priv/static/css/app.css"))
+
+    filename
+    |> Phoenix.LiveView.Utils.write_tmp_file!(".html", html)
+    |> open_browser_fun.()
+
+    GenServer.reply(from, {:ok, filename})
+    {:noreply, state}
```

Comment:
Is there any chance we can implement this on the client and not change client_proxy? That's because the logic here is slightly incorrect. Since the proxy can be out of sync with the LiveView, we should first ping the live view, and then handle this in `handle_info`. However, it will be simpler if we can implement it on the client by calling the existing `render` API.

Reply:
If you go down this route, you can make the API be this:

    open_browser(element, fun \\ &open_with_system_cmd/1)

Then we don't need a separate function for testing. :)